lib/actions.js

```diff
@@ -637,12 +637,10 @@ exports.upload = function(selector, path) {
 	var self = this;
 	if (fs.existsSync(path)) {
 		return new HorsemanPromise(function(resolve, reject) {
-			return self.page.evaluate(function(selector) {
-				self.page.uploadFile(selector, path, impatient(function() {
-					resolve();
-					debug(".upload() " + path + " into " + selector);
-				}, self.options.timeout));
-			});
+            return self.page.uploadFile(selector, path, function() {
+                resolve();
+                debug(".upload() " + path + " into " + selector);
+            });
 		});
```

Comment:
I'm sorry, do I understand it wrong: `upload` just 'binds' a path of a local computer file to the proper input field and the actual uploading process happens only after form submission?

If it's so, I don't understand why we need to mind the timeout here, because this action is just `evaluate` which does ignore the timeout.